feat: add options to output registered entity summary

paullongtan opened the pull request:
## Tracking issue
<!--
If your PR fixes an open issue, use `Closes flyteorg/flyte#999` to link your PR with the issue.
Example: Closes flyteorg/flyte#999

If your PR is related to an issue or PR, use `Related to flyteorg/flyte#999` to link your PR.
Example: Related to flyteorg/flyte#999
-->
<!-- Remove this section if not applicable -->

## Why are the changes needed?

<!--
Please clarify why the changes are needed. For instance,
1. If you propose a new API, clarify the use case for a new API.
2. If you fix a bug, you can clarify why it is a bug.
-->

## What changes were proposed in this pull request?

<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue.
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
2. If there is design documentation, please add the link.
-->

## How was this patch tested?

<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->

### Setup process

### Screenshots

## Check all the applicable boxes <!-- Follow the above conventions to check the box -->

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

## Related PRs

<!-- Add related pull requests for reviewers to check -->

## Docs link

<!-- Add documentation link built by CI jobs here, and specify the changed place -->

Mecoli1219 commented on flytekit/tools/repo.py:
```diff
@@ -350,30 +362,60 @@ def _raw_register(cp_entity: FlyteControlPlaneEntity):
+        else:
+            summary_dir = os.getcwd()
+
+        summary_file = f"registration_summary.{summary_format}"
```
Does this mean that the other registration may overwrite the summary? I think we could name the file with other unique names (ie. tmp file, py file name, wf name, version number)

paullongtan replied:
Yes! The summary will be overwritten if the registration is rerun.

Mecoli1219 commented on tests/flytekit/unit/cli/pyflyte/test_register.py:
```diff
@@ -163,3 +164,75 @@ def test_non_fast_register_require_version(mock_client, mock_remote):
+            f.close()
+        result = runner.invoke(pyflyte.main, ["register", "--summary-dir", "summaries", "core4"])
+        assert result.exit_code == 2
+        print(result.output)
```
```suggestion

```

paullongtan replied:
Done.

Mecoli1219 commented on flytekit/tools/repo.py:
```diff
@@ -350,30 +362,60 @@ def _raw_register(cp_entity: FlyteControlPlaneEntity):
                     if not skip_errors:
                         raise e
                     print_registration_status(og_id, success=False)
+                    result["status"] = "failed"
```
If it fails, what will the values of other keys be?

paullongtan replied:
The values of the other keys(id, type, version) are pre-computed before registration. Thus, the values will not be empty even if the registration fails.

The values are from:

result = {
            "id": og_id.name,
            "type": og_id.resource_type_name(),
            "version": og_id.version,
            "status": "skipped",  # default status
        }
where og_id is the id of the entity's template / entity itself.

Mecoli1219 reviewed (commented):
Nice first PR! I think you can add more unit tests.
1. yaml
2. testing registration without specifying `--summary-dir`

paullongtan commented:
> Nice first PR! I think you can add more unit tests.
> 
> 1. yaml
> 2. testing registration without specifying `--summary-dir`

Thanks! 
I have added unit tests for both yaml and registration without specifying --summary-dir.

paullongtan commented:
New updates:
1. Added option `--quiet` to pyflyte register to mute all output to CLI
2. Updated registration summary to output directly to the CLI instead of saving to a file to ensure scalability
  - uses the --quiet flag to ensure proper output format
  - drops option `--summary-dir` for pyflyte register

Note:
1. I have temporarily commented out the log for config file overwriting in `pyflyte.py` and log for packages used in `validate_package` function in `utils.py`. This is to prevent these logs from breaking the CLI output when outputting registration summary to the CLI and ensure no CLI outputs when using the `--quiet` flag for register.
